data/instruct_advanced_mysql.csv

```diff
@@ -67,14 +67,14 @@ car_dealership,mysql,instructions_date_join,"WITH RECURSIVE date_range AS (SELEC
-car_dealership,mysql,instructions_date_join,"SELECT c.make, c.model, s.sale_price FROM car_dealership.cars AS c JOIN car_dealership.sales AS s ON c.id = s.car_id JOIN car_dealership.inventory_snapshots AS i ON c.id = i.car_id AND DATE(s.sale_date) = DATE(i.snapshot_date) WHERE i.is_in_inventory = FALSE ORDER BY s.sale_price DESC LIMIT 1;","What is the make, model and sale price of the car with the highest sale price that was sold on the same day it went out of inventory?","If inventory snapshots and sales from the same day are to be joined, join on the truncated date fields eg FROM inventory_snapshots i JOIN sales s ON DATE_TRUNC('day', i.snapshot_date) = DATE_TRUNC('day', s.sale_date).","ASP (average selling price) = total sales amount / number of sales
+car_dealership,mysql,instructions_date_join,"SELECT DATE_FORMAT(p.payment_date - INTERVAL WEEKDAY(p.payment_date) DAY, '%Y-%m-%d') AS WEEK, COUNT(p.id) AS total_payments, COUNT(CASE WHEN DAYOFWEEK(p.payment_date) IN (1, 7) THEN 1 END) AS weekend_payments FROM car_dealership.payments_received AS p JOIN car_dealership.sales AS s ON p.sale_id = s.id WHERE s.sale_price > 30000 AND p.payment_date >= DATE_SUB(CURDATE(), INTERVAL 8 WEEK) AND p.payment_date < DATE_SUB(CURDATE(), INTERVAL 1 WEEK) GROUP BY WEEK ORDER BY WEEK ASC;","For sales with sale price over $30,000, how many payments were received in total and on weekends in each of the last 8 calendar weeks (excluding the current week)? Return the week (as a date), total payments received, and weekend payments received in ascending order.",Weekend days are Saturday (6) and Sunday (0). Truncate date to week for aggregation.,"To calculate the average days between sale date and payment received date, join the sales and payments received tables Weekend days are Saturday (6) and Sunday (0). Truncate date to week for aggregation. When using car makes, model names, engine_type and vin_number, match case-insensitively and allow partial matches using LIKE with wildcards. To get the total sales amount per salesperson, join the salespersons and sales tables, group by salesperson, and sum the sale_price"
```

Comment:
i think the end date of the interval is not right. it should truncate to the start of the current week instead of subtracting an entire week. So smth like `p.payment_date < DATE_SUB(CURDATE(), INTERVAL WEEKDAY(CURDATE()) DAY)`;

Reply:
looking into this!